workers/unity/Packages/io.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs

```diff
@@ -106,5 +105,32 @@ protected override void OnUpdate()
 
             EntityManager.RemoveComponent<GameObjectInitSystemStateComponent>(removedEntitiesQuery);
         }
+
+        private void ProcessNewEntities()
+        {
+            Entities.With(newEntitiesQuery).ForEach(
+                (Entity entity, ref SpatialEntityId spatialEntityId, ref Metadata.Component metadata) =>
+                {
+                    var entityType = metadata.EntityType;
+                    var expectedTypes = entityTypeExpectations.GetExpectedTypes(entityType);
+
+                    if (expectedTypes != null)
```

Comment:
If we guarantee in `ExpectedEntityTypes` that this cannot be null, we can get rid of this check. 

(I think there's any reason why this should ever be null, since an empty array is semantically equivalent to `null` here)